dashboard/app.py

```diff
@@ -68,22 +68,42 @@ def main():
 
 @app.route('/leaderboard')
 def leaderboard():
-    page_content = cache.get('leaderboard')
+    page = int(request.args.get('page', 1))
+    if page < 1:
+        abort(400)
```

Comment:
So if we have 31 nodes, and we open 7th page - we'll see empty leaderboard with "Previous" button to 6th page, which is also empty? Maybe we should return 400 or 404 in this case?

Reply:
Lets have example with 30 nodes.
Navigate from 3rd to 4th page - show empty page.
Navigate from 7th to 6th page - return 400 or 404.
How to determinate the difference between 3rd to 4th page and 7th to 6th page ?

Reply:
Ok, it's not so easy without knowing rows count. Maybe we can add this as a TODO as well?
WIth 30 nodes, 4th page should return 400 or 404, as there shouldn't be "next" link on 3rd page.
```suggestion
        abort(400)
        # TODO: return http error for non-existing pages
```

Reply:
added TODO